RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Interface.swift

```diff
@@ -111,7 +111,7 @@ extension OnLedgerEntitiesClient {
 }
 
 extension OnLedgerEntitiesClient {
-	public struct CachingStrategy: Sendable, Hashable {
+	public struct CachingStrategy: Sendable, Hashable, CustomDebugStringConvertible {
```

Comment:
We had it like that, but wanted more flexibility. There might be other dimension being added here later, i.e. expiration date, which is easier to model with a struct.

Reply:
Following your example, if we ever need an `expirationDate: Date?` property, it would only apply for the `useCache` case. Therefore, it would make more sense to have it as an associated value of the given case, than as an optional variable of the struct. With enum solution we would make sure no `CacheingStrategy` is initialized to `useCache` without an expiration date.

I know that expiration date was only an example, and you may argue that there could be new fields that apply to every case, so I'd have it like this:
<details>
  <summary>Code snippet</summary>
  
```Swift
struct CachingStrategy: String, Sendable, Hashable, CustomDebugStringConvertible {
    let kind: Kind
    let newCommonField: String
    
    enum Kind {
        case forceUpdate
        case useCache(Date)
        case readFromLedgerSkipWrite
        
        public var debugDescription: String {
            rawValue
        }
    }
    
    var read: Read {
        switch kind {
        case .forceUpdate:
            return .fromLedger
        case .useCache:
            return .fromCache
        case .readFromLedgerSkipWrite:
            return .fromLedger
        }
    }
    
    var write: Write {
        switch kind {
        case .forceUpdate:
            return .toCache
        case .useCache:
            return .toCache
        case .readFromLedgerSkipWrite:
            return .skip
        }
    }

    var date: Date? {
        switch kind {
            case let .useCache(value):
                return value
            case .forceUpdate, .readFromLedgerSkipWrite:
                return nil
        }
    }
}

extension CachingStrategy {
    enum Read: Sendable, Hashable {
        case fromCache
        case fromLedger
    }

    enum Write: Sendable, Hashable {
        case toCache
        case skip
    }
}
```
</details>

Reply:
Ah I remember now, I wanted to be able to have "presets" but still have the freedom to do e.g. `(read: false, save: true)` when debugging. I dont feel strongly about this... but lets keep it out of scope for this PR. Good that you wanna pursue enums by default, I like that!